src/tests/system/tests/test_infopipe.py

```diff
@@ -410,3 +410,196 @@ def test_infopipe__lookup_user_attributes(client: Client, provider: LDAP):
+    assert monitor_path.startswith("/org/freedesktop/sssd/infopipe/Components/"), "Invalid monitor path"
+
+    monitor = client.ifp.getObject(monitor_path)
+    props = {}
+    for prop in ["name", "type", "enabled"]:
+        props[prop] = monitor.Get("org.freedesktop.sssd.infopipe.Components", prop)
+
+    assert props["name"] == "monitor", f"Monitor name incorrect: got {props['name']}"
+    assert props["type"] == "monitor", f"Monitor type incorrect: got {props['type']}"
+    assert props["enabled"] is True, f"Monitor not enabled: got {props['enabled']}"
+
+
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_infopipe__find_user_by_id(client: Client, provider: GenericProvider):
+    """
+    :title: Test FindByID method of Users interface
+    :setup:
+        1. Add user 'iduser' with specific UID
+        2. Launch SSSD
+    :steps:
+        1. Call FindByID() with user's UID
+        2. Verify returned user object
+    :expectedresults:
+        1. Correct user path is returned
+        2. User object has expected properties
+    :customerscenario: False
+    """
+    provider.user("iduser").add(uid=10011)
+    client.sssd.start()
+
+    users = client.ifp.getObject("/org/freedesktop/sssd/infopipe/Users")
+    user_path = users.FindByID(10011)
+    assert user_path == "/org/freedesktop/sssd/infopipe/Users/test/10011", "FindByID returned wrong path"
+
+    user = client.ifp.getObject(user_path)
+    assert user.name == "iduser", "User name incorrect"
+    assert user.uidNumber == 10011, "User UID incorrect"
+
+
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_infopipe__find_group_by_id(client: Client, provider: GenericProvider):
+    """
+    :title: Test FindByID method of Groups interface
+    :setup:
+        1. Add group 'idgroup' with specific gid
+        2. Launch SSSD
+    :steps:
+        1. Call FindByID() with group's GID
+        2. Verify returned group object
+    :expectedresults:
+        1. Correct group path is returned
+        2. Group object has expected properties
+    :customerscenario: False
+    """
+    provider.group("idgroup").add(gid=20011)
+    client.sssd.start()
+
+    groups = client.ifp.getObject("/org/freedesktop/sssd/infopipe/Groups")
+    group_path = groups.FindByID(20011)
+    assert group_path == "/org/freedesktop/sssd/infopipe/Groups/test/20011", "FindByID returned wrong path"
+
+    group = client.ifp.getObject(group_path)
+    assert group.name == "idgroup", "Group name incorrect"
+    assert group.gidNumber == 20011, "Group GID incorrect"
+
+
+@pytest.mark.xfail(reason="https://issues.redhat.com/browse/IDM-1940")
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_infopipe__update_user_groups(client: Client, provider: GenericProvider):
+    """
+    :title: Call infopipe UpdateGroupsList method
+    :setup:
+        1. Create user and group, with the user as a member
```

Comment:
Since you are creating two groups: s/group/groups

Reply:
s/group/groups